spec/printer_spec.rb

```diff
@@ -207,4 +213,91 @@
       OUT
     end
   end
+
+  describe '#process_stream' do
+    subject do
+      output_io.string
+    end
+
+    around(:each) do |example|
+      Timeout::timeout(2) do
+        example.run
+      end
+    end
+
+    it 'should process a completed stream' do
```

Comment:
Nit: `should` is redundant in spec descriptions. http://www.betterspecs.org/#should